code/go/0chain.net/blobbercore/reference/ref.go

```diff
@@ -557,49 +550,27 @@ func (r *Ref) SaveDirRef(ctx context.Context) error {
 	toUpdateFileStat := r.IsPrecommit
 	prevID := r.ID
 	if r.ID > 0 {
-		err := db.Transaction(func(tx *gorm.DB) error {
-			// FIXME: temporary fix
-
-			var cnt int64
-			err := tx.Unscoped().Model(&Ref{}).Where("allocation_id=? AND path=? and deleted_at IS NOT NULL", r.AllocationID, r.Path).Count(&cnt).Error
-			if err != nil {
-				return err
-			}
-			logging.Logger.Info("SaveDirRef", zap.Any("cnt", cnt), zap.Any("path", r.Path))
-			if cnt > 0 {
-				r.IsPrecommit = true
-				err = tx.Save(r).Error
-				return err
-			}
-			err = tx.Exec("UPDATE reference_objects SET is_precommit=? WHERE id=?", false, r.ID).Error
-			if err != nil && err != gorm.ErrRecordNotFound {
-				return err
-			}
-			err = tx.Delete(&Ref{}, "id=?", r.ID).Error
-			if err != nil && err != gorm.ErrRecordNotFound {
-				return err
-			}
-			r.ID = 0
-			r.IsPrecommit = true
-			err = tx.Create(r).Error
-			if err != nil {
-				return err
-			}
-			return nil
-		})
+		err := db.Delete(&Ref{}, "id=?", r.ID).Error
```

Comment:
Is this delete related to two-phase commit?

Reply:
Yes we soft delete the previous row and create a new one to have 2 versions.